larq/optimizers_v2.py

```diff
@@ -1,10 +1,151 @@
+import warnings
+
 import tensorflow as tf
 import larq as lq
 
 from larq import utils
 from copy import deepcopy
 
 
+@utils.register_keras_custom_object
+class CaseOptimizer(tf.keras.optimizers.Optimizer):
+    """A case operation for optimizers that each train a subset of a model's variables.
+
+    An optimizer is used to train a variable iff its accompanying predicate evaluates to
```

Comment:
I am not a particular fan of the wording "an optimizer that trains a variable", maybe we should be a bit more precise.
```suggestion
    An optimizer is used to train a variable if its accompanying predicate evaluates to
```

Reply:
This "iff" was on purpose, as in [if and only if](https://en.wikipedia.org/wiki/If_and_only_if), and not a typo. Do you prefer that or just keeping it as "if"?

What don't you like about the "an optimizer that trains a variable" phrasing exactly?

Reply:
> This "iff" was on purpose, as in if and only if, and not a typo. Do you prefer that or just keeping it as "if"?

Got it

> What don't you like about the "an optimizer that trains a variable" phrasing exactly?

I personally view an optimizer as something that applies an update to a variable given a gradient signal, rather than training that variable, since for training it requires more things like data, a loss function, etc. But that's just a personal thing, since I never heard the term "training a single variable".